From 6fc34ae63294aacf18e4fb1c8909b2b957eb8db4 Mon Sep 17 00:00:00 2001
From: Olivia Lee <olivia.lee@collabora.com>
Date: Fri, 14 Nov 2025 20:29:26 -0800
Subject: [PATCH] panvk/csf: explicitly set ls_sb_slot in
 set_fbds_provoking_vertex

SB_ID(LS) is currently equal to zero, so this is not a behavior change,
but worth setting it explicitly for clarity and in case the sb
assignments change.

Signed-off-by: Olivia Lee <olivia.lee@collabora.com>
Fixes: 885805560f9 ("panvk/csf: fix case where vk_meta is used before PROVOKING_VERTEX_MODE_LAST")
Reviewed-by: Eric R. Smith <eric.smith@collabora.com>
Reviewed-by: Boris Brezillon <boris.brezillon@collabora.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/38458>
(cherry picked from commit ebbf05f9d2f98bb0d93a527f660195200fd0ba61)
---
 .pick_status.json                           | 2 +-
 src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/.pick_status.json b/.pick_status.json
index 9f179c1b8fada..1613abde37d8b 100644
--- a/.pick_status.json
+++ b/.pick_status.json
@@ -1104,7 +1104,7 @@
         "description": "panvk/csf: explicitly set ls_sb_slot in set_fbds_provoking_vertex",
         "nominated": true,
         "nomination_type": 2,
-        "resolution": 0,
+        "resolution": 1,
         "main_sha": null,
         "because_sha": "885805560f97c0b777d47db550ba51234b603a50",
         "notes": null
diff --git a/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c b/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c
index efc8dc5d90d3b..b6541d79fb2f0 100644
--- a/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c
+++ b/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c
@@ -71,6 +71,7 @@ generate_fn_set_fbds_provoking_vertex(struct panvk_device *dev,
       .nr_registers = csif_info->cs_reg_count,
       .nr_kernel_registers = MAX2(csif_info->unpreserved_cs_reg_count, 4),
       .reg_perm = provoking_vertex_fn_reg_perm_cb,
+      .ls_sb_slot = SB_ID(LS),
    };
    cs_builder_init(&b, &conf, fn_mem);
 
-- 
GitLab

